### PR TITLE
No std

### PR DESCRIPTION
**Pull request by SRCoughlin**
The `no-std` optional feature allowing for WASM build.

### CODE REVIEW

File: src/expression.rs [hunk 1]
```diff
@@ -584,6 +592,7 @@ impl<F: Field> BinaryExpression<F> {
             |sum, (i, bit)| sum + (&bit.expression * (Element::one() << i)))
     }
 
+    #[cfg(not(feature = "no-std"))]
     pub fn dependencies(&self) -> Vec<Wire> {
         let mut all = HashSet::new();
```
Comment (dlubarov) [inline src/expression.rs]: Could we use BTreeSet in both cases? I assume performance would be comparable

Comment (SRCoughlin) [reply src/expression.rs]: Agreed. I'll change.

File: src/poseidon.rs [hunk 2]
```diff
@@ -91,7 +96,7 @@ impl<F: Field> MultiPermutation<F> for Poseidon<F> {
             current = &self.mds_matrix * current.as_slice();
         }
 
-        current
+        current.to_vec()
```
Comment (dlubarov) [inline src/poseidon.rs]: Is this needed?

File: src/poseidon.rs [hunk 3]
```diff
@@ -75,7 +80,7 @@ impl<F: Field> MultiPermutation<F> for Poseidon<F> {
         assert!(self.num_rounds.full % 2 == 0, "asymmetric permutation configuration");
         let full_rounds_per_side = self.num_rounds.full / 2;
 
-        let mut current = inputs.to_owned();
+        let mut current = inputs.to_vec();//.to_owned();
```
Comment (dlubarov) [inline src/poseidon.rs]: I think `to_vec()` is fine, let's delete `to_owned()`

**Comment by dlubarov** [comment]
LGTM apart from those small nits; feel free to merge

File: src/poseidon.rs (hunk 2 above)
Comment (SRCoughlin) [reply src/poseidon.rs]: Agreed.

File: src/poseidon.rs (hunk 3 above)
Comment (SRCoughlin) [reply src/poseidon.rs]: Agreed.

File: Cargo.toml [hunk 4]
```diff
@@ -8,22 +8,22 @@ license = "MIT OR Apache-2.0"
...
 keywords = ["R1CS", "cryptography", "SNARK"]
-categories = ["cryptography"]
+categories = ["cryptography", "no-std"]
 maintenance = { status = "actively-developed" }
 edition = "2018"
 
 [features]
-default = ["bimap", "itertools", "num", "num-traits"]
-no-std = []
+default = ["std"]
```
Comment (SRCoughlin) [inline Cargo.toml]: This should be set to ```default = []``` to make the default compile `no_std` and avoid requiring the `--no-default-features` flag.

**Pull request merged**